source/AI.cpp

```diff
@@ -2854,6 +2854,31 @@ Point AI::TargetAim(const Ship &ship, const Body &target)
 	return result ? result : target.Position() - ship.Position();
 }
 
+bool AI::TargetInRange(const Ship &ship, const Hardpoint &hardpoint)
+{
+	shared_ptr<const Ship> target = ship.GetTargetShip();
+	if(target)
+		return TargetInRange(ship, hardpoint, *target);
+
+	shared_ptr<const Minable> targetAsteroid = ship.GetTargetAsteroid();
+	if(targetAsteroid)
+		return TargetInRange(ship, hardpoint, *targetAsteroid);
+
+	return false;
+}
+
+
+
+bool AI::TargetInRange(const Ship &ship, const Hardpoint &hardpoint, const Body &target)
```

Comment:
dont really see the point of making this an extra function of the AI, do you know what lambdas are?

Reply:
Didn't know c++ supported lambdas/anonymous functions/what ever name you want to use for a function in a variable.

1. My concern with turning this into a lambda would be that it limits reuse of this logic to only check for current target.
2. Where and how are you picturing the lamdba being declare? I worry that we could end up wasting cycles and memories re-declaring lamdba if we aren't careful.

If you think its just bloating the interface I can merge this and the function it overloads into one function for now.

Reply:
I was also following the overload pattern set by the AI static function `TargetAim`. But if you think its YAGNI I can just cutt it and merge the overload into the two argument function.